src/mlpack/methods/ann/loss_functions/poisson_nll_loss.hpp

```diff
@@ -0,0 +1,143 @@
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+  typename InputDataType = arma::mat,
+  typename OutputDataType = arma::mat
+>
+class PoissonNLLLoss
+{
+ public:
+  /**
+   * Create the PoissonNLLLoss object.
+   *
+   * @param logInput If true the loss is computed as exp(input) - target*input,
+   *        if false then the loss is input - target * log(input + eps).
+   * @param full Boolean value that determines whether to include Stirling's
+   *        approximation term.
+   * @param eps A small value to prevent 0 in denominators and logarithms.
+   * @param reduction When true, mean loss is computed otherwise total loss.
+   */
+  PoissonNLLLoss(const bool logInput = true,
+                 const bool full = false,
+                 const double eps = 1e-08,
+                 const bool reduction = true);
+
+  /**
+   * Computes the Poisson Negative log likelihood.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param target The target vector, that contains the class index in the range
+   *        between 1 and the number of classes.
+   */
+  template<typename InputType, typename TargetType>
+  typename InputType::elem_type Forward(const InputType& input,
+                                        const TargetType& target);
+
+  /**
+   * Ordinary feed backward pass of a neural network. The Poisson Negative Log
+   * Likelihood loss function expects the input for each class.
+   * It expects a class index, in the range between 1 and the number of classes,
+   * as target when calling the Forward function.
+   *
+   * @param input The propagated input activation.
+   * @param target The target vector, that contains the class index in the range
+   *        between 1 and the number of classes.
+   * @param output The calculated error.
+   */
+  template<typename InputType, typename TargetType, typename OutputType>
+  void Backward(const InputType& input,
+                const TargetType& target,
+                OutputType& output);
+
+  //! Get the input parameter.
+  InputDataType& InputParameter() const { return inputParameter; }
+  //! Modify the input parameter.
+  InputDataType& InputParameter() { return inputParameter; }
+
+  //! Get the output parameter.
+  OutputDataType& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the value of logInput.
+  bool LogInput() const { return logInput; }
+  //! Modify the value of logInput.
+  bool& LogInput() { return logInput; }
+
+  //! Get the value of full.
+  bool Full() const { return full; }
+  //! Modify the value of full.
+  bool& Full() { return full; }
+
+  //! Get the value of eps.
+  double Eps() const { return eps; }
+  //! Modify the value of eps.
+  double& Eps() { return eps; }
```

Comment:
I suggest to replace `logInput`, `full`, and `eps` by a couple of words.

Reply:
Do you mean more descriptive comment or more descriptive variable name? Can you clarify, please?

Reply:
I meant the comments. I had to look at the constructor's comment several times in order to read the meaning of the variables.

Reply:
Ok. Sure.
Thanks for the clarification.